### PR TITLE
Add as ROS2 package

### PR DESCRIPTION
**Pull request by Svastits**
Add ROS2 package according to [REP136](https://ros.org/reps/rep-0136.html#id13)
Directories were renamed to follow naming rules in [REP144](https://www.ros.org/reps/rep-0144.html)

### CODE REVIEW

File: kuka_external_control_sdk/changelog [hunk 1]
```diff
@@ -1,3 +1,10 @@
+kuka-external-control-sdk (1.2.1)
+
+  * Add message builder tests
+  * Make ipoc unsigned
+
+ -- Team Apollo <KUKA-HU-RD-TeamApollo@kuka.com>  Tue, 02 Apr 2024 13:12:00 +0000
```
Comment (VX792) [inline kuka_external_control_sdk/changelog:6]: Should we present the changelog here? And if so, is it okay to include a company email address even though changes could be accepted by developers outside KUKA?

File: kuka_external_control_sdk/package.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>kuka_external_control_sdk</name>
+  <version>1.2.1</version>
+  <description>Client library for KUKA external control interfaces</description>
```
Comment (VX792) [inline kuka_external_control_sdk/package.xml:6]: Maybe client libraries for external interfaces supported by KUKA? This is very minor, but I think we can emphasize that this is not a singular client library.

File: kuka_external_control_sdk_examples/package.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>kuka_external_control_sdk_examples</name>
```
Comment (VX792) [inline kuka_external_control_sdk_examples/package.xml:4]: Are we also including the examples?

File: kuka_external_control_sdk/changelog (hunk 1 above)
Comment (Svastits) [reply kuka_external_control_sdk/changelog:6]: good point, we should change to a different changelog format

File: kuka_external_control_sdk_examples/package.xml (hunk 3 above)
Comment (Svastits) [reply kuka_external_control_sdk_examples/package.xml:4]: this was needed for the industrial ci, as it tried to build all packages, but this failed unless I specified in the package.xml, that it depends on the sdk package

File: kuka_external_control_sdk/package.xml (hunk 2 above)
Comment (Svastits) [reply kuka_external_control_sdk/package.xml]: done

**Review by VX792 — APPROVED** [review]

**Pull request merged**